metricbeat/beater/metricbeat.go

```diff
@@ -5,17 +5,20 @@ import (
 	"sync"
 
 	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/publisher"
 	"github.com/elastic/beats/metricbeat/mb"
 
 	"github.com/pkg/errors"
 )
 
 // Metricbeat implements the Beater interface for metricbeat.
 type Metricbeat struct {
-	done    chan struct{}    // Channel used to initiate shutdown.
-	config  *Config          // Metricbeat specific configuration data.
-	modules []*moduleWrapper // Active list of modules.
+	done      chan struct{}    // Channel used to initiate shutdown.
+	config    *Config          // Metricbeat specific configuration data.
+	modules   []*ModuleWrapper // Active list of modules.
+	pubClient publisher.Client // Publisher client.
```

Comment:
Should we call this just client identical to the beat-generator? https://github.com/elastic/beat-generator/blob/master/%7B%7Bcookiecutter.beat%7D%7D/beater/%7B%7Bcookiecutter.beat%7D%7D.go#L19